### PR TITLE
support for armv7-unknown-linux-gnueabihf

### PR DESCRIPTION
**Pull request by morph027**
Adds support for armv7-unknown-linux-gnueabihf

refs https://github.com/signalapp/libsignal/issues/481

### CODE REVIEW

File: boring-sys/cmake/arm-linux.cmake [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR armv7)
```
Comment (jrose-signal) [inline boring-sys/cmake/arm-linux.cmake:2]: This is probably okay for now, but if someone ever wants to use the `arm-unknown-linux-gnueabihf` target to compile for ARMv6 Linux we'll have to change it, so let's rename the config file "armv7-linux" to be explicit.

**Review by jrose-signal — COMMENTED** [review]
Looks good, just one minor comment

**Comment by jrose-signal** [comment]
Also, do you have any objections to us contributing this upstream as well? They took cloudflare/boring#78, and this is pretty much an extension of that.

**Comment by morph027** [comment]
> Also, do you have any objections to us contributing this upstream as well? They took [cloudflare#78](https://github.com/cloudflare/boring/pull/78), and this is pretty much an extension of that.

Sure, upstream contributions are always good.

File: boring-sys/cmake/arm-linux.cmake (hunk 1 above)
Comment (morph027) [reply boring-sys/cmake/arm-linux.cmake:2]: Will do. Probably change the target to armv7 as well in `build.rs`.

Comment (jrose-signal) [reply boring-sys/cmake/arm-linux.cmake:2]: I think that one's correct as is; it looks like [`CARGO_CFG_TARGET_ARCH`][https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch] is different from "the architecture part of `TARGET`". We could change what the code is sniffing, but that's a bigger change.

**Comment by jrose-signal** [comment]
Guess we need to pull in upstream first…

**Comment by jrose-signal** [comment]
Rebased the libsignal branch (rebase rather than merge because it helps us keep track of what hasn't been upstreamed), then squashed and merged this in #16. Thank you for this improvement!

**Pull request closed without merging**